### DISCUSSION TITLE
Input files requierement and script

### DISCUSSION THREAD

**Discussion by JMavel** [caneparesearch/kMCpy#7] [General]
Hi! Based on a unit cell of a simple cubic nanocrystal, i'm currently trying to simulate a kmc thermal diffusion of this kind of particule over a terrasse structure made of a periodic repetition of the same particule. That's why I'm trying to write the input files required by the software. First of all, I wonder if all the files submit in the example (test/v3.nasiconbulk/input) are mandatory for the simulation and in second time how can I write the lce.json file? 
For now, I have my unit_cell cif file and also a supercell structure, I convert this unit_cell into a prim.json file in order to list all the sites available (or not), I also write by hand the initial_state.json linked to these files.
Are you able to explain to me how the lce.json file works? And have you a short explanation of each input files required ? To be sure I don't misunderstanding their meaning. 
Thks!

**Comment by dengzeyu** [comment]
Hi, thanks for the question.

The old examples were confusing because they exposed many intermediate files. In kMCpy 0.3.0, the recommended workflow is more explicit:
```shell
kmcpy sample all --output-dir kmcpy_sample
# edit kmcpy_sample/input.yaml
kmcpy run --input kmcpy_sample/input.yaml
```
Or, if you want a commented template:

```shell
kmcpy init --output input_template.yaml
```
The most important files are:

`input.yaml`: the main simulation configuration.
`structure_file`: the full template structure containing all possible sites.
`event_file`: generated event library, usually events.json.
`model_file`: kinetic model, for example local_barrier_model.json or lce.json.
`initial_state_file`: initial occupation/state file.
For simple tests, you do not need to start with an LCE. You can use a constant or rule-based local barrier model:
```python
from kmcpy.models import LocalBarrierModel

model = LocalBarrierModel.constant_barrier(300.0)  # meV
model.to("model.json")
Then generate events from your template structure:

from kmcpy.event import EventGenerator

site_mapping = {
    "Li": ["Li", "X"],
    "O": "O",
}

EventGenerator().generate_events(
    structure_file="template.cif",
    site_mapping=site_mapping,
    supercell_shape=[2, 2, 2],
    mobile_species=["Li"],
    local_env_cutoff=4.0,
    event_file="events.json",
)
```
If you really want to build an lce.json, it is not just a hand-written input file. It stores the local lattice, cluster/orbit information, and fitted parameters. A minimal construction looks like:
```python
from pymatgen.core import Structure

from kmcpy.models import LocalClusterExpansion
from kmcpy.structure import LocalLatticeStructure

structure = Structure.from_file("template.cif")

site_mapping = {
    "Li": ["Li", "X"],
    "O": "O",
}

local_lattice = LocalLatticeStructure(
    template_structure=structure,
    center=0,
    cutoff=4.0,
    site_mapping=site_mapping,
    basis_type="chebyshev",
    local_site_order="kmcpy_default",
)

lce = LocalClusterExpansion()
lce.build(
    local_lattice_structure=local_lattice,
    cutoff_cluster=[6.0, 6.0, 0.0],
)

lce.to("lce.json")
```
Then the LCE still needs fitted parameters from NEB or other calculated barrier data before it is physically meaningful.

Please try kMCpy 0.3.0 or the current development branch. Some of these helper commands and input-generation functions were recently added and are still being tested. If you can attach a minimal unit cell CIF plus the kind of diffusion event you want to model, we can help turn it into a complete working example.
